### DISCUSSION TITLE
Image preview not displayed while working on folder path project

### DISCUSSION THREAD

**Discussion by jorelart** [awcodes/filament-curator#493] [Q&A]
Anybody pleaase help me. I got an issue with curator image preview not working
Im now upload my project on some domain and path/folder, like mydomain.com/project

Here is the preview 

![image](https://github.com/awcodes/filament-curator/assets/67253819/8d4ff9f2-52b3-431e-bea6-7f524fb6ddbb)

When I inspect this, its show thumbnail url is relative on my root domain.
`<img src="/curator/media/651a23a7-2804-4561-a147-faba22edcc85.png?w=640&amp;h=320&amp;fit=crop&amp;fm=webp&amp;s=796b6b0d8dcdbac1785a860d56636f48" alt="" class="h-full object-cover w-full">`

when I add my path/folder directive on the src url above the image displayed.

How to solve this?
Thanks

**Comment by awcodes** [comment]
Make sure you APP_URL in your .env is correct.

If your Laravel app is at 'https://mydomain.com/project' then the APP_URL also needs to be 'https://mydomain.com/project'

**Reply by jorelart** [reply]
I just correct the APP_URL value but nothing changed.

I tried on local too by changing the APP_URL but when i inspect element it still show ne this `<img src="/curator/media/651a23a7-2804-4561-a147-faba22edcc85.png?w=640&amp;h=320&amp;fit=crop&amp;fm=webp&amp;s=796b6b0d8dcdbac1785a860d56636f48" alt="" class="h-full object-cover w-full">`.

Is there something i missed?

**Reply by awcodes** [reply]
That route is just a normal route in laravel. So it's based on the app url as defined by laravel.

**Reply by awcodes** [reply]
The url is correct. Something is just off in your setup or server config that is preventing the domain for the app from being the actual path it is serving the app from. This is why it's not typical, although possible, with the right nginx/apache config, to serve laravel apps in sub paths.

I would also double check that .env can't be accessed in the browser. If it is, you need to resolve that first and foremost.

**Reply by awcodes** [reply]
Here's a good [post](https://fideloper.com/laravel-subdirectory-nginx) on the topic.

**Reply by jorelart** [reply]
> The url is correct. Something is just off in your setup or server config that is preventing the domain for the app from being the actual path it is serving the app from. This is why it's not typical, although possible, with the right nginx/apache config, to serve laravel apps in sub paths.
> 
> I would also double check that .env can't be accessed in the browser. If it is, you need to resolve that first and foremost.


let me go trough on this point ☝. 

Since i use shared hosting, probably unable make any changes to the server configuration.

Just for comparison, in Livewire I also had to make changes to load assets and scripts because of working on sub paths instead of root domain.

**Reply by jorelart** [reply]
Since I couldn't find another way, I was finally forced to edit vendor resources which I shouldn't have done. 

Here i put `config('app.url')` right before curator path.

`grid-column.blade.php`

```
<div
    {{ $attributes->merge($getExtraAttributes())->class(['curator-grid-column relative rounded-t-xl overflow-hidden -mx-4 -mt-4 mb-1 aspect-video']) }}>
    @php
        $record = $getRecord();
    @endphp

    <div class="rounded-t-xl h-full overflow-hidden bg-gray-200 dark:bg-gray-700">
        @if (str($record->type)->contains('image'))
            <img src="{{ config('app.url') . $record->getSignedUrl(['w' => 640, 'h' => 320, 'fit' => 'crop', 'fm' => 'webp']) }}"
                alt="{{ $record->alt }}" @class([
                    'h-full',
                    'w-auto' => str($record->type)->contains('svg'),
                    'object-cover w-full' => !str($record->type)->contains('svg'),
                ]) />
        @else
            <x-curator::document-image :label="$record->name" icon-size="lg" :type="$record->type" :extension="$record->ext" />
        @endif
        <div
            class="absolute inset-x-0 bottom-0 flex items-center justify-between px-1.5 pt-10 pb-1.5 text-xs text-white bg-gradient-to-t from-black/80 to-transparent gap-3">
            <p class="truncate">{{ $record->pretty_name }}</p>
            <p class="flex-shrink-0">{{ $record->size_for_humans }}</p>
        </div>
    </div>
</div>
```

**Reply by awcodes** [reply]
I'm glad you have something that works. But I still think there's a bigger underlying issue outside of the plugin.

**Reply by awcodes** [reply]
What do you get if you visit https://mydomain.com/project/.env

**Reply by jorelart** [reply]
> What do you get if you visit https://mydomain.com/project/.env

Its return 403 forbidden.

**Reply by jorelart** [reply]
> I'm glad you have something that works. But I still think there's a bigger underlying issue outside of the plugin.

I hope to find the cause and fix it.

**Reply by zulmarij** [reply]
@awcodes I'm also experiencing this, has there been any progress?

**Reply by awcodes** [reply]
See the article I posted above. If you are running Laravel in a subdirectory / path then you will need to adjust your server setting to properly serve the app. This is not an issue with the plugin.

**Reply by zulmarij** [reply]
@awcodes  I ran it on example.com, but the media preview still doesn't show up. 

[This is my website nginx configuration](https://github.com/awcodes/filament-curator/discussions/333#discussioncomment-10320727)
